build/build/src/ide/web/dejavu_font.rs

```diff
@@ -0,0 +1,173 @@
+//! Downloading DejaVu Fonts.
+
+use crate::prelude::*;
+
+use enso_font as font;
+use enso_font::NonVariableDefinition;
+use enso_font::NonVariableFaceHeader;
+use ide_ci::cache::Cache;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+pub const PACKAGE_URL: &str = "https://github.com/dejavu-fonts/dejavu-fonts/releases/download/version_2_37/dejavu-fonts-ttf-2.37.zip";
+
+const PACKAGE_FONTS_PREFIX: &str = "DejaVu";
+
+const PACKAGE_SANS_MONO_PREFIX: &str = "SansMono";
+
+const DEJAVU_SANS_MONO_FONT_FAMILY_FONTS: &[(&str, font::Weight)] =
+    &[("-Bold", font::Weight::Bold), ("", font::Weight::Normal)];
+
+
+
+// ===================
+// === DejaVu Font ===
+// ===================
+
+pub async fn download_dejavu_sans_mono_font_internal(
+    cache: &Cache,
+    octocrab: &Octocrab,
+    css_basepath: Option<&str>,
+    output_path: impl AsRef<Path>,
+    css_output_path: Option<impl AsRef<Path>>,
+) -> Result {
+    let output_path = output_path.as_ref();
+    let html_fonts: Vec<_> = [
+        NonVariableFaceHeader { weight: font::Weight::Normal, ..default() },
+        NonVariableFaceHeader { weight: font::Weight::Bold, ..default() },
+    ]
+    .into_iter()
+    .collect();
+    let html_font_definitions =
+        dejavu_sans_mono_font().variations().filter(|v| html_fonts.contains(&v.header)).collect();
+    let get_font_files = async {
+        let package = get_dejavu_font_package_(cache, octocrab).await?;
+        extract_fonts(&html_font_definitions, package, output_path).await
+    };
+    let make_css_file = async {
+        if let (Some(css_basepath), Some(css_output_path)) = (css_basepath, css_output_path) {
```

Comment:
If it makes sense to pass either both `(css_basepath, css_output_path)` or neither, I'd pass an optional pair (`Option<(T,T)>`) rather than two separate options. If we want to have names, it should be a named structure.